compilers/concrete-compiler/compiler/lib/Conversion/FHETensorOpsToLinalg/TensorOpsToLinalg.cpp

```diff
@@ -1346,8 +1346,15 @@ struct ConcatRewritePattern
     llvm::ArrayRef<int64_t> outputShape = outputType.getShape();
     size_t outputDimensions = outputShape.size();
 
+    bool is_encrypted =
+        outputType.getElementType().isa<FHE::FheIntegerInterface>();
     mlir::Value result =
-        rewriter.create<FHE::ZeroTensorOp>(location, outputType).getResult();
+        is_encrypted ? rewriter.create<FHE::ZeroTensorOp>(location, outputType)
```

Comment:
Why do the elements for an encrypted tensor need to be initialized with zeros while tensors of any other type are created empty?

Reply:
It was like that before and I didn't want to change things unrelated to indexing (e.g., tests of concat).

Reply:
If all the values are rewritten, it's better to leave the tensor uninitialized in both cases. Given that encrypted values can become quite big at the lower levels, this may have an impact on performance. Also, for readability and consistency I would prefer leaving the tensor uninitialized.

Reply:
Sure.